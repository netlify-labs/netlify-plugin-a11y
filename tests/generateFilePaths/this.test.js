const path = require('path')

const PUBLISH_DIR = path.join(__dirname, 'publishDir')
const publishDir = path.relative(process.cwd(), PUBLISH_DIR)
// actual test

const pluginCore = require('../../src/pluginCore')
test('generateFilePaths works', async () => {
	const results = await pluginCore.generateFilePaths({
		absolutePublishDir: publishDir,
		fileAndDirPaths: ['/blog', '/about.html'],
		ignoreDirectories: [],
	})
	expect(results).toMatchSnapshot()
})

const pathInResults = (expectedPath, results) => {
	return results.findIndex((r) => r.endsWith(expectedPath)) != -1
}

test('ignoreDirectories works including leading slash', async () => {
	const results = await pluginCore.generateFilePaths({
		fileAndDirPaths: ['/'],
		ignoreDirectories: ['/admin'],
		absolutePublishDir: publishDir,
	})
	expect(pathInResults('publishDir/blog/post1.html', results)).toBe(true)
	expect(pathInResults('publishDir/about.html', results)).toBe(true)
	expect(pathInResults('publishDir/index.html', results)).toBe(true)
	expect(pathInResults('publishDir/admin/index.html', results)).toBe(false)
})

test('ignoreDirectories works without leading slash', async () => {
	const results = await pluginCore.generateFilePaths({
		fileAndDirPaths: ['/'],
		ignoreDirectories: ['admin'],
		absolutePublishDir: publishDir,
	})
	expect(pathInResults('publishDir/blog/post1.html', results)).toBe(true)
	expect(pathInResults('publishDir/about.html', results)).toBe(true)
	expect(pathInResults('publishDir/index.html', results)).toBe(true)
	expect(pathInResults('publishDir/admin/index.html', results)).toBe(false)
})
