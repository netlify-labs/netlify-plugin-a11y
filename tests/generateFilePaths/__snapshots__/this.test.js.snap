// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generateFilePaths works 1`] = `
Array [
  "tests/generateFilePaths/publishDir/blog/post1.html",
  "tests/generateFilePaths/publishDir/blog/post2.html",
  "tests/generateFilePaths/publishDir/about.html",
]
`;
